### PR TITLE
fix(server): validate task_id to prevent path traversal in tasks API

### PR DESCRIPTION
**Pull request by TimeToBuildBob**
Task IDs from URL parameters were used directly as filename stems in `load_task`/`save_task` without validation. A malicious task_id like `../../etc/passwd` could read/write files outside the tasks directory.

**Fix**: Add `_validate_task_id()` regex check (alphanumeric + hyphens/underscores only) to all code paths that use task_id for file I/O:
- `load_task()` — returns None for invalid IDs
- `save_task()` — raises ValueError
- `create_task_conversation()` — raises ValueError
- `setup_task_workspace()` — raises ValueError

**Tests**: 7 new tests covering parent traversal, absolute paths, dot segments, null bytes, valid IDs still work, save rejection, and API 404 response.

Part of ongoing server hardening (follows #2119, #2123, #2124).

### CODE REVIEW

File: gptme/server/tasks_api.py [hunk 1]
```diff
@@ -145,13 +146,26 @@ class TaskListResponse(BaseModel):
     tasks: list[TaskResponse] = Field(..., description="List of tasks")
 
 
+# Task IDs must be alphanumeric with hyphens/underscores, no path separators
+_TASK_ID_PATTERN = re.compile(r"^[a-zA-Z0-9_-]+$")
+
+
+def _validate_task_id(task_id: str) -> bool:
+    """Check that task_id doesn't contain path traversal characters."""
+    return bool(_TASK_ID_PATTERN.match(task_id))
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply gptme/server/tasks_api.py]: Fixed in ce0f867 — switched to `re.fullmatch` with anchors removed from the pattern. Cleaner and removes the trailing-newline edge case.

File: tests/test_tasks_api.py [hunk 2]
```diff
@@ -867,3 +867,61 @@ def test_update_valid_target_types(self, client, sample_task):
...
+            created_at="2026-01-01T00:00:00Z",
+            status="pending",
+            target_type="stdout",
+        )
+        with pytest.raises(ValueError, match="Invalid task_id"):
+            save_task(task)
+
+    def test_api_get_rejects_traversal(self, client):
+        resp = client.get("/api/v2/tasks/../../etc/passwd")
+        assert resp.status_code == 404
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply tests/test_tasks_api.py:927]: Good observation — Flask URL normalization handles the slash-traversal case before routing, so that test validates behavior (404 is correct), just not via `_validate_task_id`. The unit tests in `test_validate_task_id` exercise the function directly. I'll note this is a P2 observation and the security guarantee (no out-of-dir reads) holds regardless.

**Comment by TimeToBuildBob** [comment]
@greptileai review
